File: backend/src/utils/response.ts
import { Response } from 'express';

interface ApiResponse<T = any> {
  code: number;
  data: T;
  msg: string;
}

class ResponseUtil {
  /**
   * Success response
   */
  static success<T = any>(res: Response, data: T, msg: string = 'success') {
    const response: ApiResponse<T> = {
      code: 200,
      data,
      msg
    };
    return res.status(200).json(response);
  }

  /**
   * Created response
   */
  static created<T = any>(res: Response, data: T, msg: string = 'created') {
    const response: ApiResponse<T> = {
      code: 201,
      data,
      msg
    };
    return res.status(201).json(response);
  }

  /**
   * Bad request response
   */
  static badRequest(res: Response, msg: string = 'bad request') {
    const response: ApiResponse<null> = {
      code: 400,
      data: null,
      msg
    };
    return res.status(400).json(response);
  }

  /**
   * Unauthorized response
   */
  static unauthorized(res: Response, msg: string = 'unauthorized') {
    const response: ApiResponse<null> = {
      code: 401,
      data: null,
      msg
    };
    return res.status(401).json(response);
  }

  /**
   * Forbidden response
   */
  static forbidden(res: Response, msg: string = 'forbidden') {
    const response: ApiResponse<null> = {
      code: 403,
      data: null,
      msg
    };
    return res.status(403).json(response);
  }

  /**
   * Not found response
   */
  static notFound(res: Response, msg: string = 'not found') {
    const response: ApiResponse<null> = {
      code: 404,
      data: null,
      msg
    };
    return res.status(404).json(response);
  }

  /**
   * Internal server error response
   */
  static internalError(res: Response, msg: string = 'internal server error') {
    const response: ApiResponse<null> = {
      code: 500,
      data: null,
      msg
    };
    return res.status(500).json(response);
  }
}

export default ResponseUtil;

File: backend-go/pkg/response/response.go
package response

import (
	"net/http"

	"github.com/gin-gonic/gin"
)

type Response struct {
	Success bool        `json:"success"`
	Data    interface{} `json:"data,omitempty"`
	Message string      `json:"message,omitempty"`
	Error   string      `json:"error,omitempty"`
}

func Success(c *gin.Context, data interface{}) {
	c.JSON(http.StatusOK, Response{
		Success: true,
		Data:    data,
	})
}

func SuccessWithMessage(c *gin.Context, data interface{}, message string) {
	c.JSON(http.StatusOK, Response{
		Success: true,
		Data:    data,
		Message: message,
	})
}

func Created(c *gin.Context, data interface{}) {
	c.JSON(http.StatusCreated, Response{
		Success: true,
		Data:    data,
	})
}

func BadRequest(c *gin.Context, message string) {
	c.JSON(http.StatusBadRequest, Response{
		Success: false,
		Error:   message,
	})
}

func Unauthorized(c *gin.Context, message string) {
	c.JSON(http.StatusUnauthorized, Response{
		Success: false,
		Error:   message,
	})
}

func Forbidden(c *gin.Context, message string) {
	c.JSON(http.StatusForbidden, Response{
		Success: false,
		Error:   message,
	})
}

func NotFound(c *gin.Context, message string) {
	c.JSON(http.StatusNotFound, Response{
		Success: false,
		Error:   message,
	})
}

func InternalError(c *gin.Context, message string) {
	c.JSON(http.StatusInternalServerError, Response{
		Success: false,
		Error:   message,
	})
}

func Error(c *gin.Context, statusCode int, message string) {
	c.JSON(statusCode, Response{
		Success: false,
		Error:   message,
	})
}